src/main/java/com/volasoftware/tinder/services/implementations/FriendServiceImpl.java

```diff
@@ -63,7 +67,31 @@ public void removeFriend(Long friendId) {
     }
 
     @Override
-    public String linkRequestedAccountWithBots(Long accountId, Pageable pageable) {
+    public List<FriendDto> sortFriendsByLocation(FriendSearchDto friendSearchDto) {
+        Double accountLatitude = friendSearchDto.getLatitude();
+        Double accountLongitude = friendSearchDto.getLongitude();
+        Account loggedAccount = getLoggedAccount();
+
+        List<Account> sortedFriends = accountRepository.findFriendsByLocation(
+            loggedAccount.getId(), accountLatitude, accountLongitude);
+        if (sortedFriends.isEmpty()) {
```

Comment:
If the user doesn't have any friends, I'm not so sure that we should throw exception. I think, it's ok to return empty array. What your opinion about that?

Note: If you change to return empty array, should remove test method `testGetFriendsWhenAccountDoesNotHaveFriendsThenExceptionThrown`